### PR TITLE
DAOS-16346 tests: fix pmemcheck vea errors

### PR DESCRIPTION
**Pull request by wangshilong**
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by janekmi — DISMISSED** [review]

**Comment by janekmi** [comment]
The patch passes with pmemcheck as well.

File: src/vea/tests/vea_ut.c [hunk 1]
```diff
@@ -349,7 +345,8 @@ ut_reserve_bitmap(void **state)
 
 	/* correctness check */
 	ext = d_list_entry(r_list->prev, struct vea_resrvd_ext, vre_link);
-	assert_int_equal(ext->vre_hint_off, VEA_HINT_OFF_INVAL);
+	/* 1024 + 261 */
+	assert_int_equal(ext->vre_hint_off, 0x505);
```
Comment (NiuYawei) [inline src/vea/tests/vea_ut.c:349]: I don't see where this "1024 + 261" come from. Is it depending on previous sub-tests?

File: src/vea/tests/vea_ut.c [hunk 2]
```diff
@@ -284,10 +284,6 @@ ut_reserve_bitmap(void **state)
 	assert_rc_equal(rc, 0);
```
Comment (NiuYawei) [inline src/vea/tests/vea_ut.c:285]: Looks vea_format() is called in this sub test, I suppose ut_setup() should be called as well?

File: src/vea/tests/vea_ut.c [hunk 3]
```diff
@@ -634,8 +631,6 @@ ut_setup(struct vea_ut_args *test_args)
 	for (i = 0; i < IO_STREAM_CNT; i++) {
 		test_args->vua_hint[i] = root_addr;
 
-		test_args->vua_hint[i]->vhd_off = 0;
-		test_args->vua_hint[i]->vhd_seq = 0;
```
Comment (NiuYawei) [inline src/vea/tests/vea_ut.c:638]: To make pmemcheck happy, I think we can do the assignment in a transaction.

**Review by NiuYawei — DISMISSED** [review]

**Review by NiuYawei — APPROVED** [review]

**Review by janekmi — APPROVED** [review]

**Pull request merged**